### ISSUE TITLE
Printing doesn't work on macOS

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#2798]
When trying to do `:print` on macOS, with Qt 5.8 and 5.9, nothing happens and `QPrintDialog: Cannot be used on non-native printers` gets logged when creating `QPrintDialog` in `CommandDispatcher._print`.

In https://crashes.qutebrowser.org/view/1686e67e someone even reports a segfault when trying to print a Jupyter print preview (https://try.jupyter.org/ has examples), but I've been unable to reproduce that.

**Comment by The-Compiler** [comment]
Stacktrace I got looks related:

```
Thread 0 Crashed:: CrBrowserMain  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib         0x00007fffc7a6dd42 __pthread_kill + 10
1   libsystem_pthread.dylib       0x00007fffc7b5b457 pthread_kill + 90
2   libsystem_c.dylib             0x00007fffc7984497 raise + 26
3   libsystem_platform.dylib       0x00007fffc7b4eb3a _sigtramp + 26
4   ???                           000000000000000000 0 + 0
5   libqcocoa.dylib               0x000000011825a0f5 0x118233000 + 159989
6   QtPrintSupport                 0x0000000110fe4981 0x110fc5000 + 129409
7   PyQt5.QtPrintSupport.so       0x0000000110f90e8e sipQPrintDialog::setVisible(bool) + 110
8   QtPrintSupport                 0x0000000110fded14 QPrintDialog::open(QObject*, char const*) + 164
9   PyQt5.QtPrintSupport.so       0x0000000110f9d863 meth_QPrintDialog_open(_object*, _object*) + 259
```

**Comment by The-Compiler** [comment]
After having added a printer, I get the same crash with qutebrowser, and a print dialog but no printout with QupZilla.

**Comment by The-Compiler** [comment]
With this:

```diff
From 7c865aedbe59b02faa9f817132234a8eaa6976ba Mon Sep 17 00:00:00 2001
From: Florian Bruhin <git@the-compiler.org>
Date: Tue, 11 Jul 2017 23:41:08 +0200
Subject: [PATCH] Fix printing on macOS

Fixes #2798
---
 qutebrowser/browser/commands.py | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/qutebrowser/browser/commands.py b/qutebrowser/browser/commands.py
index 8dde90602..fac8d3a26 100644
--- a/qutebrowser/browser/commands.py
+++ b/qutebrowser/browser/commands.py
@@ -20,11 +20,12 @@
 """Command dispatcher for TabbedBrowser."""
 
 import os
+import sys
 import os.path
 import shlex
 import functools
 
-from PyQt5.QtWidgets import QApplication, QTabBar
+from PyQt5.QtWidgets import QApplication, QTabBar, QDialog
 from PyQt5.QtCore import Qt, QUrl, QEvent, QUrlQuery
 from PyQt5.QtGui import QKeyEvent
 from PyQt5.QtPrintSupport import QPrintDialog, QPrintPreviewDialog
@@ -436,9 +437,18 @@ class CommandDispatcher:
                 message.error("Printing failed!")
             diag.deleteLater()
 
+        def do_print():
+            """Called when the dialog was closed."""
+            tab.printing.to_printer(diag.printer(), print_callback)
+
         diag = QPrintDialog(tab)
-        diag.open(lambda: tab.printing.to_printer(diag.printer(),
-                                                  print_callback))
+        if sys.platform == 'darwin':
+            # For some reason we get a segfault when using open() on macOS
+            ret = diag.exec_()
+            if ret == QDialog.accepted:
+                do_print()
+        else:
+            diag.open(do_print)
 
     @cmdutils.register(instance='command-dispatcher', name='print',
                        scope='window')
-- 
2.13.2

```

there's no crash for whatever reason, but nothing gets printed either...
